conda_smithy/templates/build_steps.sh.tmpl

```diff
@@ -27,12 +27,22 @@ conda-build:
   root-dir: ${FEEDSTOCK_ROOT}/build_artifacts
 pkgs_dirs:
   - ${FEEDSTOCK_ROOT}/build_artifacts/pkg_cache
+{%- if conda_install_tool == "micromamba" %}
+  - /opt/mamba/pkgs
+{%- endif %}
   - /opt/conda/pkgs
 {%- if conda_solver %}
 solver: {{ conda_solver }}
 {%- endif %}
 
 CONDARC
+
+{%- if conda_install_tool == "micromamba" %}
+micromamba create --root-prefix /opt/conda --prefix /opt/mamba \
+    --yes --quiet --override-channels --channel conda-forge --strict-channel-priority \
+    pip {{ conda_install_tool_deps }} {{ conda_build_tool_deps }} {{ " ".join(remote_ci_setup) }}
+{%- endif %}
```

Comment:
Why can't we just install into `/opt/conda`?

Reply:
micromamba doesn't allow `--prefix` and `--root-prefix` to be the same, so this is the quickest way to reuse the `/opt/conda` cache. We also avoid `history` related problems.

If you want to reuse `/opt/conda`, we could:

1. Move `/opt/conda/pkgs` to the chosen mamba root prefix (e.g. `~/.conda`)
2. Clear `/opt/conda/conda-meta/history`
3. `micromamba install ...` into `/opt/conda`

Do you prefer that?

Reply:
Wait maybe we don't need (1). Let me check something.

Reply:
What's the issue with issue something like
```bash
CONDA_PKGS_DIRS=/opt/conda micromamba install conda-forge-ci-setup --prefix /opt/conda --root-prefix /opt/mamba
```

Reply:
Yep,  yep, that's what I tested in https://github.com/conda-forge/dav1d-feedstock/pull/21. Let me clean up and push.